### ISSUE TITLE
Commit Dialog - Stage selected Lines not working

### ISSUE THREAD

**Issue by antis81** [gitextensions/gitextensions#684]
When I select a bunch of lines in the diff view of an unstaged file and try to stage them, GitExtension says most of the time something like "error ..... patch does not apply".

Concerning this, it would be more clear, if the whole line whas selected and not just part of it.

**Comment by per1234** [comment]
The specific variant of this but as reported [previously ](https://github.com/gitextensions/gitextensions/issues/684#issuecomment-12332133)by grantbowering occurs for me when "Ignore leading and trailing whitespace changes" or "Ignore all whitespace changes" is enabled with GitExtensions 2.51.03, Windows 10 64 bit:

1. Modify a file.
2. Commands > Commit
3. Hover mouse over diff window of the commit dialog.
4. From the top toolbar that appears, enable "Ignore leading and trailing whitespace changes" and/or "Ignore all whitespace changes".
5. Click on a modified line in the diff.
6. Press the "S" key to stage the selected line: `error: patch failed:`...`error: `...`patch does not apply`.

**Comment by sylvain-bougnoux** [comment]
I've this same error for years **but on some lines not always** (for unknown reason). As said, playing with encoding or whitespace options, generally manage to bypass the error. When not, I stage the whole file and unstage the lines I don't want to commit. This time I got the error and none of these succeeded. I had to toggle "see whole file" on for success. FYI.

Windows 10
GE 4.0.2.16100
Git 2.38.1
kdiff3
